driver/bpf/plumbing_helpers.h

```diff
@@ -164,32 +210,33 @@ static __always_inline unsigned long bpf_syscall_get_argument_from_ctx(void *ctx
 		arg = 0;
 	}
 
-#else
+#elif CONFIG_S390
 	
+	/* See here for the definition:
+	 * https://github.com/libbpf/libbpf/blob/master/src/bpf_tracing.h#L132-L144
+	 */
+	user_pt_regs *user_regs = (user_pt_regs *)args->regs;
 	switch (idx) {
 	case 0:
-		arg = _READ(regs->di);
+		arg = _READ(regs->orig_gpr2);
 		break;
 	case 1:
-		arg = _READ(regs->si);
-		break;
 	case 2:
-		arg = _READ(regs->dx);
-		break;
 	case 3:
-		arg = _READ(regs->r10);
-		break;
 	case 4:
-		arg = _READ(regs->r8);
-		break;
 	case 5:
-		arg = _READ(regs->r9);
+		arg = _READ(user_regs->gprs[idx]);
```

Comment:
The first argument is placed into general-purpose register r2. Depending whether idx is zero for the first, add 2 + idx to start with gpr 2. Thanks.

Reply:
Oh silly me I have also linked the `libbpf` code with the conversion https://github.com/libbpf/libbpf/blob/master/src/bpf_tracing.h#L132-L144 🤦‍♂️ I will fix it, great catch thank you very much!!!

Reply:
No problem... that why are reviews for :-) Already started to look into this as well, also for the modern bpf parts.